### PR TITLE
Vimc 4493 refactor 2

### PR DESCRIPTION
**Pull request by xiangli313**
1. codes are running as expected
2. failed tests need to addressed
3. need to think how to deal with HepB, Measles, and Rubella which have multiple routine doses

**NOTE** this PR merges into #14, not master

### CODE REVIEW

**Review by kgaythorpe — APPROVED** [review]

File: R/impact_central.R [hunk 1]
```diff
@@ -318,25 +318,25 @@ get_birth_cohort <- function(data) {
 #' modelled. This does not account for the future disease burden averted through
 #' current vaccine activities.
 #'
-#' @param baseline_impact Data frame of baseline impact data this needs to have
+#' @param baseline_burden Data frame of baseline impact data this needs to have
```
Comment (r-ash) [inline R/impact_central.R]: Should we update the docs too?
```suggestion
#' @param baseline_burden Data frame of baseline burden data this needs to have
```

File: R/impact_central.R [hunk 2]
```diff
@@ -410,61 +410,49 @@ impact_by_birth_year <- function(baseline_impact, focal_impact) {
...
-      data.frame(matrix(ncol = 5, nrow = 0)),
-      c("country", "burden_outcome", "birth_cohort", "impact", "activity_type"))
+  if (unique(fvps$activity_type) == "campaign") {
+    raw_impact <- impact_by_calendar_year(baseline_burden,
+                                                   focal_burden)
+    names(raw_impact)[names(raw_impact) == "year"] <- "time"
+    raw_impact$activity_type <- "campaign"
   }
 
   ## Merge routine & campaign impacts
```
Comment (r-ash) [inline R/impact_central.R]:
```suggestion
```

File: R/impact_central.R [hunk 3]
```diff
@@ -480,13 +468,13 @@ impact_by_year_of_vaccination_activity_type <- function(
                                           c("fvps", "impact"))]
 
   ## Calculate impact
-  fvps_country <- stats::aggregate(fvps ~ country + activity_type + year,
+  fvps_country <- stats::aggregate(fvps ~ country + vaccine + activity_type + year,
```
Comment (r-ash) [inline R/impact_central.R:470]: If `vaccine` is required let's add a validation check to call to assert_has_columns. Will external users be coming with data for different vaccines? Should this disaggregation be used in other impact methods?

File: R/impact_central.R [hunk 4]
```diff
@@ -550,13 +538,13 @@ impact_by_year_of_vaccination_birth_cohort <- function(
                                           c("fvps", "impact"))]
 
   ## Calculate impact
-  fvps_country <- stats::aggregate(fvps ~ country + birth_cohort + year,
+  fvps_country <- stats::aggregate(fvps ~ country + birth_cohort + year + vaccine + activity_type,
```
Comment (r-ash) [inline R/impact_central.R:540]: Let's add this to the `assert_has_columns` check above

**Review by r-ash — CHANGES_REQUESTED** [review]
This is great Xiang, thanks for fixing up that impact method. Couple of small comments

File: R/impact_central.R (hunk 1 above)
Comment (xiangli313) [reply R/impact_central.R]: You are right. We should update those. Will do after our meeting.

**Review by r-ash — APPROVED** [review]

**Pull request merged**